code/modules/mob/living/carbon/human/human.dm

```diff
@@ -1092,6 +1092,16 @@
 
 	if(usr == src)
 		self = 1
+
+	if (isipc(src)) 
+		usr << span("notice", self ? "You are a machine, you do not have a pulse." : "[src] is a machine, they do not have a pulse.")
+		return
+
+	// just in-case
+	if (src.species.flags & NO_BLOOD)

```

Comment:
NO_BLOOD covers isipc, so no need for specific case if general case already catches it. Also, maybe rework the message to say that they don't have a heartbeat as a species. Saying they don't have blood is odd.